### PR TITLE
Add s3 bucket resource

### PR DESCRIPTION
**Pull request by nvthongswansea**
Changes:
- Add s3 bucket resource.
- Add s3 bucket resource docs.
- go mod tidy & vendor.

### CODE REVIEW

File: website/docs/r/bucket.html.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+---
+
+# gridscale_object_storage_accesskey
+
+Provides a bucket. This can be used to create, and delete buckets.
+
+## Example Usage
+
+```terraform
+resource "gridscale_object_storage_accesskey" "foo" {
```
Comment (itakouna) [inline website/docs/r/bucket.html.md:16]: should it not be `gridscale-resource-bucket`?

File: website/docs/r/bucket.html.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+layout: "gridscale"
+page_title: "gridscale: bucket"
+sidebar_current: "docs-gridscale-resource-bucket"
+description: |-
+   Manages a bucket in gridscale.
+---
+
+# gridscale_object_storage_accesskey
```
Comment (itakouna) [inline website/docs/r/bucket.html.md:9]: same here?

File: gridscale/provider.go [hunk 3]
```diff
@@ -106,6 +106,7 @@ func Provider() *schema.Provider {
 			"gridscale_marketplace_application":        resourceGridscaleMarketplaceApplication(),
 			"gridscale_marketplace_application_import": resourceGridscaleImportedMarketplaceApplication(),
 			"gridscale_ssl_certificate":                resourceGridscaleSSLCert(),
+			"gridscale_bucket":                         resourceGridscaleBucket(),
```
Comment (itakouna) [inline gridscale/provider.go]: would it not be better to use `s3_bucket` or `object_storage_bucket` instead of the `bucket`?

Comment (nvthongswansea) [reply gridscale/provider.go]: Done @itakouna

**Review by itakouna — APPROVED** [review]

**Pull request merged**